sycl/doc/design/CompileTimeProperties.md

```diff
@@ -111,32 +111,32 @@ template </* ... */> class
 } // namespace sycl::ext::oneapi
 ```
 
-The device compiler only uses the
-`[[__sycl_detail__::add_ir_attributes_global_variable()]]` attribute when the
-decorated type is used to create an [LLVM IR global variable][3] and the global
-variable's type is either:
+When the device compiler creates an [LLVM IR global variable][3] of the 
+annotated class it will use the
+`[[__sycl_detail__::add_ir_attributes_global_variable()]]` attribute to generate
+entries in the `@llvm.global.annotations` list, as described in
+[IR representation via `@llvm.global.annotations`][5].
```

Comment:
Won't this break backwards compatibility?

Reply:
Technically yes. Though this is our own implementation detail, and as far as I am aware it is only used on `device_globals` which are in the `experimental` namespace which is allowed to break backward compatibility.